### PR TITLE
[ENH] Move expriemental surface maskers to stable part of the code base 

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #

Changes proposed in this pull request:

-
-

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Note to self so I remember when coming back from holidays.

Revert last code changes (not the doc string ones) so this PR is just a "move". Should ensure that tests pass. Address compliance with sklearn estimators in follow up PRs.

**Comment by Remi-Gau** [comment]
Will split this PR in 2: one for surface masker and one for surface label masker to make it easier to review

File: nilearn/maskers/surface_masker.py [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+    ----------
+    output_dimension_: :obj:`int` or None
+        number of vertices included in mask
+
+    mask_img_: SurfaceImage or None
+    """
+
+    def __init__(
+        self,
+        mask_img=None,
```
Comment (Remi-Gau) [inline nilearn/maskers/surface_masker.py:79]: extra arguments that exist for NiftiMasker that could be introduced later:
 - mask_strategy: strategy used to compute the mask
 - mask_args: additional parameters passed to masking.compute_background_mask or masking.compute_epi_mask to fine-tune mask computation

may require masking.py to work on surface data

Comment (bthirion) [reply nilearn/maskers/surface_masker.py:79]: Not so sure we want to have them here.  These functions are meant to extract automatically a brain mask from EPI data. I don't really see what it would bring on a brain surface mesh.

**Review by bthirion — COMMENTED** [review]
THe PR looks good but:
1. There are some codecov complaints.
2. the corresponding code has not been removed from experimental ?

File: nilearn/maskers/surface_masker.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/maskers/surface_masker.py:79]: hum... yeah I think this may not make much sense...

**Comment by Remi-Gau** [comment]
> 2\. the corresponding code has not been removed from experimental ?

Thanks. I kind of forgot that important part when splitting the PR in 2: the number of line changes should have been a tell.

**Comment by Remi-Gau** [comment]
> 1. codecov complaints

I checked coverage locally and I do not see the codecov complaints

File: nilearn/maskers/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+
+from nilearn.maskers._utils import (
+    compute_mean_surface_image,
+    get_min_max_surface_image,
+)
+
+
+def test_compute_mean_surface_image(surf_img, assert_surf_img_equal):
```
Comment (bthirion) [inline nilearn/maskers/tests/test_utils.py:10]: one-line docstring would be great

File: nilearn/maskers/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    print(input_img.shape[0])
+    img = compute_mean_surface_image(two_time_points_img)
+
+    assert_array_almost_equal(
+        img.data.parts["left"], np.squeeze(np.ones(shape=(1, 4)) * 0.5)
+    )
+
+
+def test_get_min_max_surface_image(surf_img):
```
Comment (bthirion) [inline nilearn/maskers/tests/test_utils.py:33]: Same here

**Review by bthirion — COMMENTED** [review]
Codecov says 76% patch coverage...

**Comment by Remi-Gau** [comment]
> Codecov says 76% patch coverage...

I think codecov is wrong somehow.

When running this on my branch:

```bash
pytest nilearn/maskers/tests --cov=nilearn/maskers/ --cov-report=term 

```

I get a better coverage for the surface masker than for several other masker modules.

```
---------- coverage: platform linux, python 3.12.3-final-0 -----------
Name                                           Stmts   Miss Branch BrPart  Cover
--------------------------------------------------------------------------------
nilearn/maskers/__init__.py                       10      0      0      0   100%
nilearn/maskers/_utils.py                         34      0     16      0   100%
nilearn/maskers/base_masker.py                    73     11     26      3    80%
nilearn/maskers/multi_nifti_labels_masker.py      29      0      8      1    97%
nilearn/maskers/multi_nifti_maps_masker.py        29      0      8      1    97%
nilearn/maskers/multi_nifti_masker.py             88      8     40      9    85%
nilearn/maskers/nifti_labels_masker.py           251     75     88      6    70%
nilearn/maskers/nifti_maps_masker.py             193     60     70      3    67%
nilearn/maskers/nifti_masker.py                  153     40     50      2    72%
nilearn/maskers/nifti_spheres_masker.py          230     66     74      6    70%
nilearn/maskers/surface_masker.py                143      9     46      8    91%
--------------------------------------------------------------------------------
TOTAL                                           1233    269    426     39    76%
```

File: nilearn/maskers/tests/test_utils.py (hunk 2 above)
Comment (Remi-Gau) [reply nilearn/maskers/tests/test_utils.py:10]: done

File: nilearn/maskers/tests/test_utils.py (hunk 3 above)
Comment (Remi-Gau) [reply nilearn/maskers/tests/test_utils.py:32]: done

**Review by bthirion — APPROVED** [review]
LGTM, thx.
